### ISSUE TITLE
Getting compilation error in examples build

### ISSUE THREAD

**Issue by rinkalbhatia06** [sysrepo/sysrepo#2143]
![image](https://user-images.githubusercontent.com/69560210/92857212-ff1e0400-f411-11ea-9595-95fa71812c63.png)

**Comment by rinkalbhatia06** [comment]
If I comment //#include "compat.h" (Line number=25) in file application_changes_example.c, then it compiles successfully.

**Comment by michalvasko** [comment]
Hi,
what *sysrepo* version are you compiling? I am not sure how this can happen with the latest versions.

**Comment by rinkalbhatia06** [comment]
![image](https://user-images.githubusercontent.com/69560210/92877561-e705af00-f428-11ea-9c05-0ffa169371a0.png)

**Comment by michalvasko** [comment]
Hm, okay, can you please test this patch? Thanks.
```patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 15184918..97708792 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -187,6 +187,7 @@ set (SYSREPOPLUGIND_SRC
 
 # use compat
 use_compat()
+include_directories(${PROJECT_BINARY_DIR})
 
 # sysrepo
 set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
```

**Comment by rinkalbhatia06** [comment]
After applying patch
examples/build
make output:
![image](https://user-images.githubusercontent.com/69560210/92898400-93e92780-f43b-11ea-8819-ecab399be1ec.png)

**Comment by michalvasko** [comment]
Uh, right, please test this patch instead
```patch
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 2235fe9d..5a46eb7f 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -8,7 +8,7 @@ set(examples application_changes_example sr_set_item_example sr_get_items_exampl
     notif_send_example rpc_send_example rpc_subscribe_example oper_pull_push_example)
 
 foreach(app_name IN LISTS examples)
-    add_executable(${app_name} ${app_name}.c)
+    add_executable(${app_name} ${app_name}.c $<TARGET_OBJECTS:compat>)
     target_link_libraries(${app_name} sysrepo)
 endforeach(app_name)
 
```
and if it still does not work, then also add
```cmake
include_directories(${PROJECT_BINARY_DIR})
```
at the end of this `CMakeLists.txt` and let me know whether it works then.

**Comment by rinkalbhatia06** [comment]
I have tried your patch 

This error is coming using git
error: corrupt patch at line 13

**Comment by michalvasko** [comment]
Hi,
so I assume you did apply the patch. Why it does not work for you at all I do not know, but I think there is something broken on your side. I tried it on Ubuntu 18.04 VM (which you are probably using) and failed to reproduce this with any of my patches. In any case, I have removed `compat` from this example because it was fairly easy to do and it was not linked to the example anyway. So, it should work for you now, but I still think something else may easily break.
